viewmodels/src/commonMain/kotlin/com/mirego/trikot/viewmodels/ToggleSwitchViewModel.kt

```diff
@@ -12,4 +12,9 @@ interface ToggleSwitchViewModel : ViewModel {
      * Set the state defined by the platform switch
      */
     fun setIsOn(on: Boolean)
+
+    /**
+     * If the ToggleSwitch is enabled or disabled
+     */
+    val enabled: Publisher<Boolean>
```

Comment:
I think it's a good timing to introduce the concept of `ControlViewModel` to abstract ViewModels that can be enabled/disabled just like apple does with UIControls (https://developer.apple.com/documentation/uikit/uicontrol). 

We would then have :
```kotlin
interface ControlViewModel: ViewModel {
    /**
     * A Boolean value indicating whether the control is in the enabled or disabled state.
     */
    val enabled: Publisher<Boolean>
}
```

Then our ButtonViewModel, ToggleViewModel and InputTextViewModel could implement this Control interface instead of repeating this attribute.

Reply:
I agree, I had the same debate for ScratchViewModels, but android devs did not agree at the time.